### Project: Exploring Ebay Car Sales Data

The aim of the project is to munge and analyze car listings from Ebay.

The data dictionary provided with data is as follows:

- dateCrawled - When this ad was first crawled. All field-values are taken from this date.
- name - Name of the car.
- seller - Whether the seller is private or a dealer.
- offerType - The type of listing
- price - The price on the ad to sell the car.
- abtest - Whether the listing is included in an A/B test.
- vehicleType - The vehicle Type.
- yearOfRegistration - The year in which the car was first registered.
- gearbox - The transmission type.
- powerPS - The power of the car in PS.
- model - The car model name.
- kilometer - How many kilometers the car has driven.
- monthOfRegistration - The month in which the car was first registered.
- fuelType - What type of fuel the car uses.
- brand - The brand of the car.
- notRepairedDamage - If the car has a damage which is not yet repaired.
- dateCreated - The date on which the eBay listing was created.
- nrOfPictures - The number of pictures in the ad.
- postalCode - The postal code for the location of the vehicle.
- lastSeenOnline - When the crawler saw this ad last online.

## Import modules

In [1]:
import pandas as pd

### Import Data

In [2]:
autos = pd.read_csv("autos.csv", encoding = "Windows-1252")

In [3]:
# Review the data
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [4]:
autos.head(3)

dateCrawled                                        name  seller  \
0  2016-03-26 17:47:46            Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1  2016-04-04 13:38:56  BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2  2016-03-26 18:57:24                  Volkswagen_Golf_1.6_United  privat   

  offerType   price   abtest vehicleType  yearOfRegistration    gearbox  \
0   Angebot  $5,000  control         bus                2004    manuell   
1   Angebot  $8,500  control   limousine                1997  automatik   
2   Angebot  $8,990     test   limousine                2009    manuell   

   powerPS   model   odometer  monthOfRegistration fuelType       brand  \
0      158  andere  150,000km                    3      lpg     peugeot   
1      286     7er  150,000km                    6   benzin         bmw   
2      102    golf   70,000km                    7   benzin  volkswagen   

  notRepairedDamage          dateCreated  nrOfPictures  postalCode  \
0              nein  2016-03-26 00:00:00             0       79588   
1              nein  2016-04-04 00:00:00             0       71034   
2              nein  2016-03-26 00:00:00             0       35394   

              lastSeen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37

Looking at the first few rows of data, it appears that there is some wrangling that can take place. For example, in the second column, name, the company and make/model of the car is in a single cell. We should probably parse that out so we can explore the data further... for example, if we wanted to see all the BMW statistics.

In [5]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

Some other observations, there is missing data, for example in the vehicleType column and in the gearbox column. Also, the price is a text column and not a numeric column.

In [6]:
autos.describe(include = 'all')

dateCrawled         name  seller offerType  price abtest  \
count                 50000        50000   50000     50000  50000  50000   
unique                48213        38754       2         2   2357      2   
top     2016-03-27 22:55:05  Ford_Fiesta  privat   Angebot     $0   test   
freq                      3           78   49999     49999   1421  25756   
mean                    NaN          NaN     NaN       NaN    NaN    NaN   
std                     NaN          NaN     NaN       NaN    NaN    NaN   
min                     NaN          NaN     NaN       NaN    NaN    NaN   
25%                     NaN          NaN     NaN       NaN    NaN    NaN   
50%                     NaN          NaN     NaN       NaN    NaN    NaN   
75%                     NaN          NaN     NaN       NaN    NaN    NaN   
max                     NaN          NaN     NaN       NaN    NaN    NaN   

       vehicleType  yearOfRegistration  gearbox       powerPS  model  \
count        44905        50000.000000    47320  50000.000000  47242   
unique           8                 NaN        2           NaN    245   
top      limousine                 NaN  manuell           NaN   golf   
freq         12859                 NaN    36993           NaN   4024   
mean           NaN         2005.073280      NaN    116.355920    NaN   
std            NaN          105.712813      NaN    209.216627    NaN   
min            NaN         1000.000000      NaN      0.000000    NaN   
25%            NaN         1999.000000      NaN     70.000000    NaN   
50%            NaN         2003.000000      NaN    105.000000    NaN   
75%            NaN         2008.000000      NaN    150.000000    NaN   
max            NaN         9999.000000      NaN  17700.000000    NaN   

         odometer  monthOfRegistration fuelType       brand notRepairedDamage  \
count       50000         50000.000000    45518       50000             40171   
unique         13                  NaN        7          40                 2   
top     150,000km                  NaN   benzin  volkswagen              nein   
freq        32424                  NaN    30107       10687             35232   
mean          NaN             5.723360      NaN         NaN               NaN   
std           NaN             3.711984      NaN         NaN               NaN   
min           NaN             0.000000      NaN         NaN               NaN   
25%           NaN             3.000000      NaN         NaN               NaN   
50%           NaN             6.000000      NaN         NaN               NaN   
75%           NaN             9.000000      NaN         NaN               NaN   
max           NaN            12.000000      NaN         NaN               NaN   

                dateCreated  nrOfPictures    postalCode             lastSeen  
count                 50000       50000.0  50000.000000                50000  
unique                   76           NaN           NaN                39481  
top     2016-04-03 00:00:00           NaN           NaN  2016-04-07 06:17:27  
freq                   1946           NaN           NaN                    8  
mean                    NaN           0.0  50813.627300                  NaN  
std                     NaN           0.0  25779.747957                  NaN  
min                     NaN           0.0   1067.000000                  NaN  
25%                     NaN           0.0  30451.000000                  NaN  
50%                     NaN           0.0  49577.000000                  NaN  
75%                     NaN           0.0  71540.000000                  NaN  
max                     NaN           0.0  99998.000000                  NaN

Looking at some summary statistics for the columns reveals some work to be done. There are quite a few columns that only have two unique values. They may be candidates to drop. An example is seller, below: 

In [7]:
autos["seller"].unique()

array(['privat', 'gewerblich'], dtype=object)

## Data Munging

First thing we want to do is change the column names to the preferred style from camelcase to snakecase.

In [8]:
old_names = autos.columns
old_names

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [9]:
new_names = ['date_crawled','name','seller','offer_type', 'price','ab_test','vehicle_type',
            'registration_year', 'gear_box','power_ps', 'model', 'odometer','registration_month',
            'fuel_type', 'brand','unrepaired_damage', 'ad_created', 'number_of_pics', 'postal_code',
            'last_seen']

In [10]:
# Change old column names to new names
autos.columns = new_names

In [11]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gear_box', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'number_of_pics', 'postal_code',
       'last_seen'],
      dtype='object')

In [12]:
# Change price column to numeric by removing '$' and ','
def convert_columns(col):
    col = (col.str.replace('$','')
           .str.replace(',','')
           .str.replace('km','')
           .astype(float)
          )
    return col
autos['price'] = convert_columns(autos['price'])

In [13]:
# Convert odometer column to numeric and rename it
autos['odometer'] = convert_columns(autos['odometer'])

autos.rename(columns = {'odometer':'odometer_km'}, inplace = True)

In [17]:
# View the result
autos[["price","odometer_km"]].head(3)

price  odometer_km
0  5000.0     150000.0
1  8500.0     150000.0
2  8990.0      70000.0

In [18]:
# Now we explore the price and odometer_km to see if there are other issues
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [21]:
autos['price'].unique().shape

(2357,)

In [22]:
autos['price'].value_counts()

0.0           1421
500.0          781
1500.0         734
2500.0         643
1200.0         639
1000.0         639
600.0          531
800.0          498
3500.0         498
2000.0         460
999.0          434
750.0          433
900.0          420
650.0          419
850.0          410
700.0          395
4500.0         394
300.0          384
2200.0         382
950.0          379
1100.0         376
1300.0         371
3000.0         365
550.0          356
1800.0         355
5500.0         340
1250.0         335
350.0          335
1600.0         327
1999.0         322
              ... 
2225.0           1
69997.0          1
139997.0         1
69999.0          1
4780.0           1
8930.0           1
21599.0          1
15911.0          1
10000000.0       1
5180.0           1
919.0            1
1247.0           1
5998.0           1
27020.0          1
21888.0          1
46500.0          1
2001.0           1
2459.0           1
345000.0         1
34940.0          1
2785.0           1
5248.0      

There are definitely some more problems with the price column, for example there are 1421 entries with a price of 0. There is also a vehicle that cost 10,000,000. That could be an outlier, but it might not be. I decided to remove it after a quick google search because the closest car I could find was about GBP700,000 which is significantly different than 10,000,000.

In [32]:
autos[autos['price'] == 10000000]

date_crawled                                               name  \
11137  2016-03-29 23:52:57  suche_maserati_3200_gt_Zustand_unwichtig_laufe...   

       seller offer_type       price  ab_test vehicle_type  registration_year  \
11137  privat    Angebot  10000000.0  control        coupe               1960   

      gear_box  power_ps model  odometer_km  registration_month fuel_type  \
11137  manuell       368   NaN     100000.0                   1    benzin   

                brand unrepaired_damage           ad_created  number_of_pics  \
11137  sonstige_autos              nein  2016-03-29 00:00:00               0   

       postal_code            last_seen  
11137        73033  2016-04-06 21:18:11

In [35]:
# Remove price outliers
autos = autos[(autos['price'] > 0) & (autos['price'] < 10000000)]

In [38]:
# Now to look at the odometer_km column
autos['odometer_km'].unique().shape

(13,)

In [39]:
autos['odometer_km'].value_counts()

150000.0    31417
125000.0     5058
100000.0     2115
90000.0      1734
80000.0      1415
70000.0      1217
60000.0      1155
50000.0      1013
5000.0        837
40000.0       815
30000.0       780
20000.0       762
10000.0       253
Name: odometer_km, dtype: int64

It doesn't look like there is any issues with the odometer_km. There are a bunch of cars with low km's, but that is common when you see classic cars for sale as well as cars that people only own for a year.

Now to explore date columns to see if there are any issues. Currently the data is stored as strings for the date_crawled, last_seen and the ad_created columns.

In [40]:
autos[['date_crawled','ad_created','last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

In [51]:
# Explore distribution of date_crawled
(autos["date_crawled"]
 .str[:10].
 value_counts(normalize=True, dropna=False)
 .sort_index()
)

2016-03-05    0.025324
2016-03-06    0.014041
2016-03-07    0.036009
2016-03-08    0.033291
2016-03-09    0.033086
2016-03-10    0.032180
2016-03-11    0.032571
2016-03-12    0.036915
2016-03-13    0.015668
2016-03-14    0.036544
2016-03-15    0.034280
2016-03-16    0.029606
2016-03-17    0.031644
2016-03-18    0.012909
2016-03-19    0.034774
2016-03-20    0.037883
2016-03-21    0.037389
2016-03-22    0.033003
2016-03-23    0.032221
2016-03-24    0.029338
2016-03-25    0.031603
2016-03-26    0.032200
2016-03-27    0.031089
2016-03-28    0.034856
2016-03-29    0.034115
2016-03-30    0.033683
2016-03-31    0.031830
2016-04-01    0.033683
2016-04-02    0.035474
2016-04-03    0.038603
2016-04-04    0.036524
2016-04-05    0.013094
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64

This is a hard way to view a distribution, but plotting isn't in the course yet. So, looking at the data above it appears that the last date in the data set is probably not a complete day but otherwise it looks reasonable

In [56]:
(autos["ad_created"]
 .str[:10]
 .value_counts(normalize=True, dropna=False)
 .sort_index()
)

2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
2015-12-30    0.000021
2016-01-03    0.000021
2016-01-07    0.000021
2016-01-10    0.000041
2016-01-13    0.000021
2016-01-14    0.000021
2016-01-16    0.000021
2016-01-22    0.000021
2016-01-27    0.000062
2016-01-29    0.000021
2016-02-01    0.000021
2016-02-02    0.000041
2016-02-05    0.000041
2016-02-07    0.000021
2016-02-08    0.000021
2016-02-09    0.000021
2016-02-11    0.000021
2016-02-12    0.000041
2016-02-14    0.000041
2016-02-16    0.000021
2016-02-17    0.000021
2016-02-18    0.000041
2016-02-19    0.000062
2016-02-20    0.000041
2016-02-21    0.000062
                ...   
2016-03-09    0.033147
2016-03-10    0.031891
2016-03-11    0.032900
2016-03-12    0.036750
2016-03-13    0.017006
2016-03-14    0.035186
2016-03-15    0.034012
2016-03-16    0.030121
2016-03-17    0.031294
2016-03-18    0.013588
2016-03-19    0.033683
2016-03-20    0.037944
2016-03-21 

There appears to be very few adds created before March 9th.

In [57]:
# last_seen column
(autos["last_seen"]
 .str[:10].
 value_counts(normalize=True, dropna=False)
 .sort_index()
)

2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005394
2016-03-08    0.007412
2016-03-09    0.009594
2016-03-10    0.010665
2016-03-11    0.012374
2016-03-12    0.023780
2016-03-13    0.008894
2016-03-14    0.012600
2016-03-15    0.015874
2016-03-16    0.016450
2016-03-17    0.028083
2016-03-18    0.007350
2016-03-19    0.015832
2016-03-20    0.020650
2016-03-21    0.020630
2016-03-22    0.021371
2016-03-23    0.018530
2016-03-24    0.019765
2016-03-25    0.019209
2016-03-26    0.016800
2016-03-27    0.015647
2016-03-28    0.020877
2016-03-29    0.022359
2016-03-30    0.024768
2016-03-31    0.023780
2016-04-01    0.022791
2016-04-02    0.024933
2016-04-03    0.025200
2016-04-04    0.024480
2016-04-05    0.124786
2016-04-06    0.221799
2016-04-07    0.131931
Name: last_seen, dtype: float64

Nothing odd jumps out at me in this column.

In [58]:
# Review the registration year
autos['registration_year'].describe()

count    48571.000000
mean      2004.754257
std         88.638548
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

There is definitely some weirdness in the registration_year column. For example, there is a max year of 9999 (will we have teleporters instead of cars then???) and a min of 1000... before we had cars. Another issue to look out for is you cannot have a car registered outside of the ad posting date. For example, if the care went on sale in 2016, you can't have a registration year of 2017. For the analysis, I will set an acceptable range of values of 1920 - 2016.

In [60]:
# Remove autos outside our acceptable range
autos = autos[(autos['registration_year'] > 1919) & (autos['registration_year'] <= 2016)]

### Exploring the brand Column

Next we will explore the brand column. I'm doing this so I can aggregate by brand and explore the top brands.

In [67]:
brand_mean_prices = {}
brands = autos['brand'].unique()

for brand in brands:
    brand_mean_prices[brand] = autos.loc[autos['brand'] == brand, 'price'].mean()

In [68]:
brand_mean_prices

{'alfa_romeo': 4087.690322580645,
 'audi': 9336.687453600594,
 'bmw': 8571.480147917478,
 'chevrolet': 6684.139097744361,
 'chrysler': 3465.743902439024,
 'citroen': 3779.1391437308866,
 'dacia': 5915.528455284553,
 'daewoo': 1049.0,
 'daihatsu': 1636.1965811965813,
 'fiat': 2813.748538011696,
 'ford': 4054.6930147058824,
 'honda': 4107.857923497268,
 'hyundai': 5365.254273504273,
 'jaguar': 11635.493150684932,
 'jeep': 11650.5,
 'kia': 5982.330303030303,
 'lada': 2688.296296296296,
 'lancia': 3376.22,
 'land_rover': 19108.091836734693,
 'mazda': 4112.596614950635,
 'mercedes_benz': 8628.450366422385,
 'mini': 10613.459657701711,
 'mitsubishi': 3394.5729166666665,
 'nissan': 4743.40252454418,
 'opel': 2976.3217131474103,
 'peugeot': 3094.0172290021537,
 'porsche': 45643.93706293706,
 'renault': 2475.7172727272728,
 'rover': 1602.2903225806451,
 'saab': 3211.6493506493507,
 'seat': 4397.230949589683,
 'skoda': 6368.0,
 'smart': 3580.2239031770046,
 'sonstige_autos': 23618.856209150326,


Using a dictionary makes it difficult to see the top brands. However, it appears that Porsche has the hightest mean value of all of them.

Exploring the mean mileage is up next and converting the mean_prices dictionary into a dataframe

In [69]:
brand_mean_mileage = {}
for brand in brands:
    brand_mean_mileage[brand] = autos.loc[autos['brand'] == brand, 'odometer_km'].mean()

In [70]:
brand_mean_mileage

{'alfa_romeo': 131338.70967741936,
 'audi': 129157.38678544914,
 'bmw': 132575.9050214091,
 'chevrolet': 99191.72932330827,
 'chrysler': 132378.0487804878,
 'citroen': 119694.18960244648,
 'dacia': 84268.29268292683,
 'daewoo': 121642.85714285714,
 'daihatsu': 116410.2564102564,
 'fiat': 117121.9715956558,
 'ford': 124266.23774509804,
 'honda': 122172.13114754099,
 'hyundai': 106442.30769230769,
 'jaguar': 124178.08219178082,
 'jeep': 127122.64150943396,
 'kia': 112530.30303030302,
 'lada': 83518.51851851853,
 'lancia': 121900.0,
 'land_rover': 118010.20408163265,
 'mazda': 124464.03385049365,
 'mercedes_benz': 130788.36331334666,
 'mini': 88105.13447432763,
 'mitsubishi': 126575.52083333333,
 'nissan': 118330.99579242637,
 'opel': 129311.75298804781,
 'peugeot': 127153.62526920316,
 'porsche': 96853.14685314686,
 'renault': 128127.27272727272,
 'rover': 137661.29032258064,
 'saab': 144415.58441558442,
 'seat': 121131.30128956624,
 'skoda': 110848.5639686684,
 'smart': 99326.7776096822

In [78]:
bmp = pd.Series(brand_mean_prices)
bmm = pd.Series(brand_mean_mileage)
bm = pd.DataFrame(bmp, columns=['mean_prices'])
bm['mean_mileage'] = bmm

bm

mean_prices   mean_mileage
alfa_romeo       4087.690323  131338.709677
audi             9336.687454  129157.386785
bmw              8571.480148  132575.905021
chevrolet        6684.139098   99191.729323
chrysler         3465.743902  132378.048780
citroen          3779.139144  119694.189602
dacia            5915.528455   84268.292683
daewoo           1049.000000  121642.857143
daihatsu         1636.196581  116410.256410
fiat             2813.748538  117121.971596
ford             4054.693015  124266.237745
honda            4107.857923  122172.131148
hyundai          5365.254274  106442.307692
jaguar          11635.493151  124178.082192
jeep            11650.500000  127122.641509
kia              5982.330303  112530.303030
lada             2688.296296   83518.518519
lancia           3376.220000  121900.000000
land_rover      19108.091837  118010.204082
mazda            4112.596615  124464.033850
mercedes_benz    8628.450366  130788.363313
mini            10613.459658   88105.134474
mitsubishi       3394.572917  126575.520833
nissan           4743.402525  118330.995792
opel             2976.321713  129311.752988
peugeot          3094.017229  127153.625269
porsche         45643.937063   96853.146853
renault          2475.717273  128127.272727
rover            1602.290323  137661.290323
saab             3211.649351  144415.584416
seat             4397.230950  121131.301290
skoda            6368.000000  110848.563969
smart            3580.223903   99326.777610
sonstige_autos  23618.856209   89869.281046
subaru           3953.290000  125300.000000
suzuki           4096.732852  108465.703971
toyota           5167.091062  115944.350759
trabant          1818.609375   55312.500000
volkswagen       5604.071269  128711.476075
volvo            4946.501171  138067.915691